src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -91,7 +100,34 @@ protected void maybeSetLargeIcon(NotificationData data) {
 
     protected void maybeSetExpandableNotification(NotificationData data) {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
-            builder.setStyle(new Notification.BigTextStyle().bigText(data.message));
+            if (null != data.expandableImageUrl && data.expandableImageUrl.startsWith("http")) {
+                try {
+                    Bitmap imageBitmap = getBitmapFromUrl(data.expandableImageUrl);
+
+                    if (imageBitmap == null) {
+                        setBigTextStyle(data.message);
```

Comment:
Let's add some debug logging for every possible flow to help developers debug. Here's an example:
```java
MPLog.d(LOGTAG, "getBitmapFromUrl returned null, setting BigTextStyle instead. Image URL: " + data.expandableImageUrl);
```